xarray/tests/test_range_index.py

```diff
@@ -14,11 +14,51 @@ def create_dataset_arange(
     return xr.Dataset(coords=xr.Coordinates.from_xindex(index))
 
 
-def test_range_index_arange() -> None:
-    index = RangeIndex.arange(0.0, 1.0, 0.1, dim="x")
+@pytest.mark.parametrize(
+    "args,kwargs",
+    [
+        ((10.0,), {}),
+        ((), {"stop": 10.0}),
+        (
+            (
+                2.0,
+                10.0,
+            ),
+            {},
+        ),
+        ((2.0,), {"stop": 10.0}),
+        ((), {"start": 2.0, "stop": 10.0}),
+        ((2.0, 10.0, 2.0), {}),
+        ((), {"start": 2.0, "stop": 10.0, "step": 2.0}),
+    ],
+)
+def test_range_index_arange(args, kwargs) -> None:
+    index = RangeIndex.arange(*args, **kwargs, dim="x")
+    actual = xr.Coordinates.from_xindex(index)
+    expected = xr.Coordinates({"x": np.arange(*args, **kwargs)})
+    assert_equal(actual, expected, check_default_indexes=False)
+
+
+def test_range_index_arange_error() -> None:
+    with pytest.raises(TypeError, match=".*requires stop to be specified"):
+        RangeIndex.arange(dim="x")
+
+
+def test_range_index_arange_start_as_stop() -> None:
+    # Weird although probably very unlikely case where only `start` is given
```

Comment:
pretty common IME :(

Reply:
Do you refer to `arange(10)` or `arange(start=10)`? This test and comment is for the latter only, which seems pretty unusual to me but I might be wrong!

Reply:
Ah Ok I misunderstood the comment then.